client/ayon_core/hosts/max/api/lib.py

```diff
@@ -584,6 +584,29 @@ def get_tyflow_export_particle_operators(members):
     return opt_list
 
 
+def get_tyflow_export_operators():
+    opt_list = []
+    event_filter = {}
+    members = [obj for obj in rt.Objects if rt.ClassOf(obj) == rt.tyFlow]
+    for member in members:
+        obj = member.baseobject
+        anim_names = rt.GetSubAnimNames(obj)
+        for anim_name in anim_names:
+            sub_anim = rt.GetSubAnim(obj, anim_name)
+            if not rt.isKindOf(sub_anim, rt.tyEvent):
+                continue
+            node_names = rt.GetSubAnimNames(sub_anim)
+            event_filter[str(sub_anim)] = []
+            for node_name in node_names:
+                node_sub_anim = rt.GetSubAnim(sub_anim, node_name)
+                if rt.hasProperty(node_sub_anim, "exportMode"):
+                    element_sets = [node_sub_anim]
+                    if not event_filter.get(str(sub_anim), None):
+                        event_filter[str(sub_anim)] = element_sets
+            opt_list.extend(event_filter[str(sub_anim)])
+    return opt_list
```

Comment:
There doesn't seem to be a reason these are put into the `event_filter` since you're not using the `event_filter` as output for anything.
You do seem to filter against whether any elements were collected - if not, then collect it. But if that's what you want, why not break out of the loop on the first one found?


```suggestion
def get_tyflow_export_operators():
    """Return all TyFlow operators with `exportMode` properties
    
    Returns:
        List[object]: List of TyFlow Events.
    """
    
    operators = []
    members = [obj for obj in rt.Objects if rt.ClassOf(obj) == rt.tyFlow]
    for member in members:
        obj = member.baseobject
        anim_names = rt.GetSubAnimNames(obj)
        for anim_name in anim_names:
            sub_anim = rt.GetSubAnim(obj, anim_name)
            if not rt.isKindOf(sub_anim, rt.tyEvent):
                continue
            node_names = rt.GetSubAnimNames(sub_anim)
            for node_name in node_names:
                node_sub_anim = rt.GetSubAnim(sub_anim, node_name)
                if not rt.hasProperty(node_sub_anim, "exportMode"):
                    continue
                    
                # We found the sub anim export node for this sub anim.
                operators.append(node_sub_anim)
                break

    return operators
```


Or if you prefer this for readability:
```suggestion
def get_tyflow_export_operators():
    """Return all TyFlow operators with `exportMode` properties
    
    Returns:
        List[object]: List of TyFlow Events.
    """
    
    def _get_sub_anim_event_export_node(sub_anim):
        for name in rt.GetSubAnimNames(sub_anim):
            node = rt.GetSubAnim(sub_anim, name)
            if not rt.hasProperty(name, "exportMode"):
                continue
            return node
    
    operators = []
    members = [obj for obj in rt.Objects if rt.ClassOf(obj) == rt.tyFlow]
    for member in members:
        obj = member.baseobject
        anim_names = rt.GetSubAnimNames(obj)
        for anim_name in anim_names:
            sub_anim = rt.GetSubAnim(obj, anim_name)
            if not rt.isKindOf(sub_anim, rt.tyEvent):
                continue
                
            export_node = _get_sub_anim_event_export_node(sub_anim)
            if export_node:
                operators.append(export_node)

    return operators
```

Note that I do not know much about TyFlow, so there may be reasons to do what you're doing - but from reading the code I don't really understand why it's done that way. :)

Reply:
maybe i shouldnt push the change to this PR, but i am stilll in a work in progress to the creator so some of the logic in this function would be changed like eventually